backend/dal/Services/Concrete/BuildingService.cs

```diff
@@ -147,7 +147,7 @@ public Building Get(int id)
                     && (isAdmin
                         || b.IsVisibleToOtherAgencies
                         || ((!b.IsSensitive || viewSensitive)
-                            && userAgencies.Contains(b.AgencyId)))) ?? throw new KeyNotFoundException();
+                            && (userAgencies.Contains(b.AgencyId) || userAgencies.Contains(b.Agency.ParentId))))) ?? throw new KeyNotFoundException();
```

Comment:
This doesn't look valid.  The user's claims should include all agencies they have access to.  Ensure your claims are correct for the user you are testing.

Reply:
Ah must of been how I had a user set up locally, thank you

Reply:
> Wouldn't you also need to update the `ParcelService` if there was an issue adding buildings to other agencies? Additionally would you also need to update all the `Update` methods?

Update looks like it has the correct implementation, checks if `userAgencies` contains the original agency or they are an admin. Which should be fine if the users claims include all agencies they have access to. But I do need to add it to `ParcelService.cs`